designs/NuGet-CPPCLI-Support.md

```diff
@@ -0,0 +1,114 @@
+
+To understand the proposed changes, here's a mapping of all `C++` project types and the property value for the appropriate ones.
+
+| Project type | TargetFrameworkIdentifier | TargetPlatformIdentifier | CLRSupport | Effective NuGet framework | Notes | Open questions |
+|--------------|---------------------------|--------------------------|------------|---------------------------|-------|----------------|
+| Native C++ | .NETFramework,Version=v4.0 | Windows,Version=10.0.19041.0 | false | native | NuGet will continue special casing vcxproj. |
+| C++ UWP App | | UAP,Version=10.0.18362.0 | false | native | NuGet will continue special casing vcxproj. |
+| CLR C++ | .NETFramework,Version=v4.7.2 | Windows,Version=10.0.19041.0 | true | ??? | This project has .NET Framework CLR support | Are these projects expected to be PackageReference or packages.config? What's the effective target framework |
+| Core CLR C++ | .NETCoreApp,Version=v5.0 | Windows,Version=10.0.19041.0 | NetCore | ??? | What's the effective framework? Is it `net5.0-windows`? |
+
+In the background, it is said that C++/CLI projects are supposed to support installing both managed and native packages. This would require some amendments to NuGet's framework model.
+
+`native` framework is not compatible with anything but `native` and that will remain.
+NuGet has a fallback compatibility mode as well.
+For example: .NET Core projects have a fallback to .NET Framework with a warning. This is also callled `Asset Target Fallback`.
+If a package supports .NET Core or .NET Standard, it's fully compatible with a .NET Core project. Otherwise if a package has .NET Framework assets, the package is installed, but with a *warning*.
+
+The `Asset Target Fallback` implementation currently suffers from a bug where the dependencies are not pulled in correctly, see [5957](https://github.com/NuGet/Home/issues/5957).
+
+Whatever the effective frameworks turn out to be, there are 2 approaches that we can take.
+
+* Utilize AssetTargetFallback
+
+This would require project type changes. The project type would specify which target framework to fall back to. This likely requires fixing [5957](https://github.com/NuGet/Home/issues/5957).
+Note that this would *generate* a warning when managed packages are used.
+
+* Introduce a `dedicated dual compatibility` framework
+
+Similar to how `.NET Core` implements `.NET Standard` and supports `.NET Core` as a same family framework, we can add a framework type that's only allowed in projects, and not packages. This framework could define it's compatible as both `native` and `managed`.
+
+The downside here is that NuGet does not have a framework that's only supported in projects. Given that this is not a real `target`, it is counter to what NuGet frameworks are supposed to represent.
+
+## Test Strategy
+
+TBD
+
+## Future Work

```

Comment:
APIs for how NuGet is supposed to inform the C++ project system about modifications on package install/uninstall/upgrade?

Reply:
You mean Package Manager UI changes?
That'll go through `UnconfiguredProject`, same for managed/SDK-based CPS PR projects.